misc-scripts/xref_mapping/XrefParser/ProcessData.pm

```diff
@@ -101,21 +101,23 @@ DSS
   my $dep_sth = $dbi->prepare($sql);
 
   # validate species names
-  if (defined $division) { push @$rspecies, $division; }
-  my @species_ids = $self->validate_species($rspecies, $verbose);
-  if (defined $taxon_id) { push @species_ids, $taxon_id; }
+  my $species_id = $self->validate_species($species, $verbose);
+  my $division_id = $self->validate_species($division, $verbose);
+  my @species_sources = ($species_id);
+  push @species_sources, $division_id if defined $division_id;
+  push @species_sources, $taxon_id if defined $taxon_id;
 
   # validate source names
-  exit(1) if ( !$self->validate_sources(\@species_ids,$sources, $verbose) );
-  exit(1) if ( !$self->validate_sources(\@species_ids,$notsources, $verbose) );
+  exit(1) if ( !$self->validate_sources(\@species_sources,$sources, $verbose) );
+  exit(1) if ( !$self->validate_sources(\@species_sources,$notsources, $verbose) );
 
   # build SQL
   my $species_sql = "";
-  if (@species_ids) {
+  if (@species_sources) {
     $species_sql .= " AND su.species_id IN (";
```

Comment:
$species_sql initialised to "" at line 115.  If it enters the if block, the resulting $species_sql might not be a valid SQL.